sequencing-server/src/lib/codegen/CommandTypeCodegen.ts

```diff
@@ -355,21 +349,43 @@ function mapArgumentType(argument: ampcs.FswCommandArgument, enumMap: ampcs.Enum
   }
 }
 
-export async function processDictionary(dictionary: ampcs.CommandDictionary) {
-  const prefaceUrl = new URL('./CommandEDSLPreface.ts', import.meta.url);
-  const jsonSpecUrl = new URL('../../../node_modules/@nasa-jpl/seq-json-schema/types.ts', import.meta.url);
+export async function processDictionary(
+  dictionary: ampcs.CommandDictionary | ampcs.ChannelDictionary | ampcs.ParameterDictionary,
+  type: 'COMMAND' | 'CHANNEL' | 'PARAMETER' = 'COMMAND',
+): Promise<string> {
   const folder = `${getEnv().STORAGE}/${dictionary.header.mission_name.toLowerCase()}/`;
-  const fileName = `command_lib.${dictionary.header.version}.ts`;
-  const prefaceString = await fs.promises.readFile(prefaceUrl.pathname, 'utf8');
-  const jsonSpecString = `/** START Sequence JSON Spec */
+
+  switch (type) {
+    case 'COMMAND': {
+      const prefaceUrl = new URL('./CommandEDSLPreface.ts', import.meta.url);
+      const jsonSpecUrl = new URL('../../../node_modules/@nasa-jpl/seq-json-schema/types.ts', import.meta.url);
+      const fileName = `command_lib.${dictionary.header.version}.ts`;
+      const prefaceString = await fs.promises.readFile(prefaceUrl.pathname, 'utf8');
+      const jsonSpecString = `/** START Sequence JSON Spec */
   //https://github.com/NASA-AMMOS/seq-json-schema/blob/develop/types.ts\n
   ${await fs.promises.readFile(jsonSpecUrl.pathname, 'utf8')}
   \n/** END Sequence JSON Spec */`;
 
-  const { values, declarations } = generateTypescriptCode(dictionary); //update sql table store seprate
+      const { values, declarations } = generateTypescriptCode(dictionary as ampcs.CommandDictionary); //update sql table store seprate
 
-  await fs.promises.mkdir(folder, { recursive: true });
+      await fs.promises.mkdir(folder, { recursive: true });
 
-  await fs.promises.writeFile(folder + fileName, prefaceString + jsonSpecString + declarations + values, { flag: 'w' });
-  return folder + fileName;
+      await fs.promises.writeFile(folder + fileName, prefaceString + jsonSpecString + declarations + values, {
+        flag: 'w',
+      });
+      return folder + fileName;
+    }
+    case 'CHANNEL': {
+      const fileName = `channel_lib.${dictionary.header.version}.ts`;
+      await fs.promises.mkdir(folder, { recursive: true });
+      await fs.promises.writeFile(folder + fileName, JSON.stringify(dictionary as ChannelDictionary), { flag: 'w' });
+      return folder + fileName;
+    }
+    case 'PARAMETER': {
```

Comment:
If we're going with this standard we should probably tie these values to a real `enum` rather than just using magic strings.

Reply:
Good idea, I created one.